### PR TITLE
Re-allow `-fPIC` compiler flag during compilation

### PR DESCRIPTION
**Pull request by sbc100**
We were temporally disabling this flag due to issues with PIC code
in static linking.  These issues should not be resolved.

Fixes: #9013

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -8377,7 +8377,6 @@ def test_undefined_main(self):
 
   def test_fpic_static(self):
     self.emcc_args.append('-fPIC')
-    self.emcc_args.remove('-Werror')
```
Comment (kripken) [inline tests/test_core.py]: why is this changed?

Comment (sbc100) [reply tests/test_core.py]: Because we no long print a warning when `fPIC` is used.

**Comment by sbc100** [comment]
The chrome failure is unrelated to this change so I'm landing with admin privs.
